datasource/common/src/main/java/org/opennms/oce/datasource/common/ImmutableAlarm.java

```diff
@@ -0,0 +1,227 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.datasource.common;
+
+import java.util.Objects;
+
+import org.opennms.oce.datasource.api.Alarm;
+import org.opennms.oce.datasource.api.Severity;
+
+public final class ImmutableAlarm implements Alarm {
+    private final String id;
+    private final long time;
+    private final Severity severity;
+    private final String inventoryObjectId;
+    private final String inventoryObjectType;
+    private final String summary;
+    private final String description;
+
+    private ImmutableAlarm(Builder builder) {
+        this.id = builder.id;
+        this.time = builder.time;
+        this.severity = builder.severity;
+        this.inventoryObjectId = builder.inventoryObjectId;
+        this.inventoryObjectType = builder.inventoryObjectType;
+        this.summary = builder.summary;
+        this.description = builder.description;
+    }
+
+    public static class Builder {
+        private String id;
+        private long time;
+        private Severity severity;
+        private String inventoryObjectId;
+        private String inventoryObjectType;
+        private String summary;
+        private String description;
+
+        private Builder() {
+            time = System.currentTimeMillis();
+        }
+
+        private Builder(Alarm alarm) {
+            this.id = alarm.getId();
+            this.time = alarm.getTime();
+            this.severity = alarm.getSeverity();
+            this.inventoryObjectId = alarm.getInventoryObjectId();
+            this.inventoryObjectType = alarm.getInventoryObjectType();
+            this.summary = alarm.getSummary();
+            this.description = alarm.getDescription();
+        }
+
+        public String getId() {
```

Comment:
Is there any need to include the getters in the builder? Given that we're build the object, we must already know what value we set.

Reply:
In most cases, no not necessary. However I added them for consistency because I was using them for Situations (for blacklisting).

However, I think there is an easier way to do that so I should be able to get rid of all the getters.